staging/src/k8s.io/apimachinery/pkg/util/diff/compare.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 Authors of Arktos.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diff
+
+import "time"
+
+var allowedNTPDiffInMilliSecond = uint64(5000)
+
+// For backwards compatible - can still use ETCD revision # starting from 1 and increases by 1 as backend storage
+var v2MinRevision = getRevisionNumber(time.Date(2020, 5, 10, 0, 0, 0, 0, time.UTC).UnixNano(), 0, 1)
+
+// RevisionIsNewer is used in event comparison to check whether revision 1 is newer than
+// revision 2 and should be sent/accepted/processed
+func RevisionIsNewer(revision1, revision2 uint64) bool {
+	if revision1 < v2MinRevision || revision2 < v2MinRevision {
```

Comment:
I suppose that reversion less than v2MinRevision  is the old one before we make the change. In that case, if revision1 < v2MinRevision  while  revision2 > v2MinRevision(revision2 is newer than revision1), the return result is revision1 newer than revision2  in case revision1 > revision2?

Reply:
yes